pkg/strategy/production.go

```diff
@@ -71,7 +75,23 @@ func newProductionStrategy(ctx context.Context, jaeger *v1.Jaeger) S {
 		if q := route.NewQueryRoute(jaeger).Get(); nil != q {
 			c.routes = append(c.routes, *q)
 		}
+
+		// For openshift platform, we want the clusterIP service to be annotated
+		// with the cert secret-name tag
+		for k := range c.services {
+			span.AddEvent(ctx, "Checking service", core.KeyValue{Key: "Servicename", Value: core.String(c.services[k].Name)})
+			if c.services[k].Name == service.GetNameForHeadlessCollectorService(jaeger) {
+				if c.services[k].Annotations == nil {
+					c.services[k].Annotations = map[string]string{
+						"service.beta.openshift.io/serving-cert-secret-name": fmt.Sprintf("%s-tls", c.services[k].Name),
```

Comment:
I wonder if it wouldn't be easier to add this directly to the relevant service? We might not even need to check if the platform is openshift, as Kubernetes will just ignore this unknown annotation...

https://github.com/jaegertracing/jaeger-operator/blob/9d6319d0f0e828e2a1781d7d5f6a0369d15439c8/pkg/service/collector.go#L33-L78

Reply:
You're right, but its just a small extra check to add it to the service on the openshift platform. It could also confuse k8s users to see that annotation